flink-table/flink-table-common/src/test/java/org/apache/flink/table/catalog/CatalogPropertiesUtilTest.java

```diff
@@ -0,0 +1,159 @@
+                        final UnresolvedPhysicalColumn expectedColumn =
+                                (UnresolvedPhysicalColumn) expected.getColumns().get(i);
+                        assertThat(actualColumn.getName()).isEqualTo(expectedColumn.getName());
+                        assertThat(actualColumn.getDataType().toString())
+                                .isEqualTo(expectedColumn.getDataType().toString());
+                    }
+                };
+        schemaComparision.accept(deserializedModel.getInputSchema(), catalogModel.getInputSchema());
+        schemaComparision.accept(
+                deserializedModel.getOutputSchema(), catalogModel.getOutputSchema());
+
+        assertThat(deserializedModel.getOptions()).isEqualTo(catalogModel.getOptions());
+        assertThat(deserializedModel.getComment()).isEqualTo(catalogModel.getComment());
+    }
+
+    @Test
+    public void testCatalogTableSerde() {
+        final Map<String, String> options = new HashMap<>();
+
+        final CatalogTable catalogTable =
+                CatalogTable.newBuilder()
+                        .schema(
+                                Schema.newBuilder()
+                                        .column(
+                                                "f1",
+                                                DataTypes.INT()
+                                                        .getLogicalType()
+                                                        .asSerializableString())
+                                        .column(
+                                                "f2",
+                                                DataTypes.STRING()
+                                                        .getLogicalType()
+                                                        .asSerializableString())
+                                        .primaryKey("f1")
+                                        .build())
+                        .comment("some comment")
+                        .options(options)
+                        .build();
+
+        final Column f1 = Column.physical("f1", DataTypes.INT());
+        final Column f2 = Column.physical("f2", DataTypes.STRING());
+        List<Column> columns = Arrays.asList(f1, f2);
+        final UniqueConstraint primaryKey =
+                UniqueConstraint.primaryKey("PK_f1", Collections.singletonList("f1"));
+        final ResolvedSchema schema =
+                new ResolvedSchema(columns, Collections.emptyList(), primaryKey);
+
+        final ResolvedCatalogTable testTable = new ResolvedCatalogTable(catalogTable, schema);
+
+        final Map<String, String> serializedMap =
+                CatalogPropertiesUtil.serializeCatalogTable(testTable);
+        final CatalogTable deserializedTable =
+                CatalogPropertiesUtil.deserializeCatalogTable(serializedMap);
+
+        // We need to compare the fields one by one since equalsTo in UnresolvedDataType in
```

Comment:
nit: you can also simplify the code by comparing `schema.toString`. The toString is stable and used at a couple of places already. This should make this method very short.